spec/models/event_stream_spec.rb

```diff
@@ -21,4 +21,38 @@
       end
     end
   end
+
+  context "description" do
+    it "raises NotImplementedError, for subclasses to implement" do
+      expect {described_class.description}.to raise_error(NotImplementedError)
+    end
+  end
+
+  context "group_and_level_details" do
+    it "raises NotImplementedError, for subclasses to implement" do
+      expect {described_class.group_and_level_details}.to raise_error(NotImplementedError)
+    end
+  end
+
+  it "doesn't support timeline" do
+    expect(described_class.new.supports?(:timeline)).to be_falsey
+  end
+
+  context "event_stream_options" do
+    it "works" do
+      MiqEventDefinitionSet.seed
+      options = described_class.event_stream_options
+      expect(options.keys.sort).to eq %w[EmsEvent MiqEvent]
+
+      expect(options["EmsEvent"].keys.sort).to eq %w[description group_levels group_names]
+      expect(options["EmsEvent"]["description"]).to eq(EmsEvent.description)
+      expect(options["EmsEvent"]["group_levels"].keys.sort).to eq %w[critical detail warning]
+      expect(options["EmsEvent"]["group_names"].keys).to include *%i[addition configuration console deletion devices]
+
+      expect(options["MiqEvent"].keys.sort).to eq %w[description group_levels group_names]
+      expect(options["MiqEvent"]["description"]).to eq(MiqEvent.description)
+      expect(options["MiqEvent"]["group_levels"].keys.sort).to eq %w[failure success]
```

Comment:
Does it make sense to `to_i` all nested keys to have consistent keys since the event_handling part of the settings.yml has symbols keys, or should the nested keys be strings to match how they'd be accessed over json?

Reply:
Do you mean `to_s` all of the keys?  If so, I'm not sure.

Reply:
Yeah, I think I'll just use the existing settings.yml event_handling format and do symbols keys even though there's no such concept on the other side of json.

Reply:
Either way, I don't like that there's currently both string and symbol keys based on what level of depth.

Reply:
FYI, this is what I meant... I changed it in a commit I just pushed:

    Change mixed string / symbol keys to all symbol keys

    Before:

    ```
    {"EmsEvent"=>
      {"description"=>"Management Events",
       "group_names"=>
        {:addition=>"Creation/Addition",
    ...
         :security=>"Security"},
       "group_levels"=>
        {"critical"=>"Critical", "detail"=>"Detail", "warning"=>"Warning"}},
     "MiqEvent"=>
      {"description"=>"Policy Events",
       "group_names"=>
        {"vm_process"=>"VM Lifecycle",
    ...
         "auth_validation"=>"Authentication Validation (Provider)"},
       "group_levels"=>{"success"=>"Success", "failure"=>"Failure"}}}
    ```

    After:

    ```
    {:EmsEvent=>
      {:description=>"Management Events",
       :group_names=>
        {:addition=>"Creation/Addition",
    ...
         :security=>"Security"},
       :group_levels=>
        {:critical=>"Critical", :detail=>"Detail", :warning=>"Warning"}},
     :MiqEvent=>
      {:description=>"Policy Events",
       :group_names=>
        {:vm_process=>"VM Lifecycle",
    ...
         :auth_validation=>"Authentication Validation (Provider)"},
       :group_levels=>{:success=>"Success", :failure=>"Failure"}}}
    ```